src/app/api/native/git/branches/route.ts

```diff
@@ -0,0 +1,294 @@
+// src/app/api/native/git/branches/route.ts
+import { NextRequest, NextResponse } from 'next/server';
+import * as git from 'isomorphic-git';
+import fs from 'fs';
+import path from 'path';
+
+// Get the repository path from the environment variable
+const LOCAL_TAXONOMY_ROOT_DIR = process.env.NEXT_PUBLIC_LOCAL_TAXONOMY_ROOT_DIR || `${process.env.HOME}/.instructlab-ui`;
+
+interface Diffs {
+  file: string;
+  status: string;
+}
+
+export async function GET() {
+  const REPO_DIR = path.join(LOCAL_TAXONOMY_ROOT_DIR, '/taxonomy');
+  try {
+    // Ensure the repository path exists
+    if (!fs.existsSync(REPO_DIR)) {
+      return NextResponse.json({ error: 'Repository path does not exist.' }, { status: 400 });
+    }
+
+    // List all branches in the repository
+    const branches = await git.listBranches({ fs, dir: REPO_DIR });
+    const branchDetails = [];
+
+    for (const branch of branches) {
+      const branchCommit = await git.resolveRef({ fs, dir: REPO_DIR, ref: branch });
+      const commitDetails = await git.readCommit({ fs, dir: REPO_DIR, oid: branchCommit });
+
+      branchDetails.push({
+        name: branch,
+        creationDate: commitDetails.commit.committer.timestamp * 1000 // Convert to milliseconds
+      });
+    }
+
+    branchDetails.sort((a, b) => b.creationDate - a.creationDate); // Sort by creation date, newest first
+
+    console.log('Branches present in native taxonomy:', branchDetails);
+    return NextResponse.json({ branches: branchDetails }, { status: 200 });
+  } catch (error) {
+    console.error('Failed to list branches:', error);
+    return NextResponse.json({ error: 'Failed to list branches' }, { status: 500 });
+  }
+}
+
+// Handle POST requests for merge or branch comparison
+export async function POST(req: NextRequest) {
+  const LOCAL_TAXONOMY_DIR = path.join(LOCAL_TAXONOMY_ROOT_DIR, '/taxonomy');
+  const { branchName, action, remoteTaxonomyRepoDir } = await req.json();
+  console.log('Received POST request:', { branchName, action, remoteTaxonomyRepoDir });
+
+  if (action === 'delete') {
+    return handleDelete(branchName, LOCAL_TAXONOMY_DIR);
+  }
+
+  if (action === 'diff') {
+    return handleDiff(branchName, LOCAL_TAXONOMY_DIR);
+  }
+
+  if (action === 'publish') {
+    return handlePublish(branchName, LOCAL_TAXONOMY_DIR, remoteTaxonomyRepoDir);
+  }
+  return NextResponse.json({ error: 'Invalid action specified' }, { status: 400 });
+}
+
+async function handleDelete(branchName: string, localTaxonomyDir: string) {
+  try {
+    if (!branchName || branchName === 'main') {
```

Comment:
Should we also check against other protected branches?  Since im not sure if we care about this I will not repeat the comment for every instance in the file, but there are multiple

Reply:
UI workflows are triggered from the contributions, and all the contributions are created using it's own branch. Ideally branchName === 'main' condition should not even hit in UI case, so we can even remove this check, but kept it as a guardrails. In local taxonomy, main is the only protected branch at this point of time.